### PR TITLE
[Docs] Update `vizro-core` docs to meet Vale's standards

### PR DESCRIPTION
**Pull request by stichbury**
## Description
* Few minor tweaks to Vale styles
* Few changes to vizro-ai docs that linting with vale spotted in addition to the changes already made and merged for #381 
* Mostly changes a number of vizro-core docs pages. Not that I've not yet edited most of the user-guides pages readability, so at this point I have ONLY tweaked them to get them to pass Vale standards. (Changes such as fixing capitalisation, headings, removing some words like "Please" and "Simply" and similar).

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

  - I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
  - I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorized to submit this contribution on behalf of the original creator(s) or their licensees.
  - I certify that the use of this contribution as authorized by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
  - I have not referenced individuals, products or companies in any commits, directly or indirectly.
  - I have not added data or restricted code in any commits, directly or indirectly.

### CODE REVIEW

**Comment by stichbury** [comment]
Now ready for review @huong-li-nguyen @maxschulz-COL 

I'm confused as to why the `--strict` flag is causing the docs build to break on this branch (because of changes introduced in the data manager docs additions) but why it _didn't_ break the docs build for the PR that introduced it. Any ideas? And it's not breaking it for `main`. It's odd!

I can fix those issues here but because I don't understand why they're occurring in this singular branch, just want to find out what's happening first.

File: vizro-core/docs/pages/user-guides/data.md
Comment (maxschulz-COL) [inline vizro-core/docs/pages/user-guides/data.md:1]: This is the source of the rendering issues! And this is actually a bit worrying as it seems to overwrite the most recent version of the docs with an outdated one, so maybe some merge must have gone wrong.

If we were to use `main` as is, the docs build just fine.

We should definitely check what changes are intended, and where have things gone wrong somewhere

**Review by maxschulz-COL — COMMENTED** [review]
The source of the docs building failure seems to be a wrong merge, see comment above.

**Comment by stichbury** [comment]
Closing this PR as the branch didn't merge correctly with updates to main

**Pull request closed without merging**